pkg/apply/apply.go

```diff
@@ -42,12 +44,37 @@ func (a *Applicator) ApplyWorkflowWithHieraData(workflowName string, hieraData m
 
 func (a *Applicator) applyWithHieraData(workflowName string, hieraData map[string]string, intent wfapi.Operation) {
 	tp := func(ic lookup.ProviderContext, key string, _ map[string]eval.Value) (eval.Value, bool) {
-		v, ok := hieraData[key]
-		return types.WrapString(v), ok
+		m := convertToDeepMap(hieraData)
```

Comment:
This is very inefficient. Do the conversion on line 46 instead (outside of this func). Otherwise, the conversion takes place once for every lookup.

Reply:
👍

Reply:
Change made.